### PR TITLE
Convert more tests to Kotlin

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

**Comment by yschimke** [comment]
Failures look relevant

```
 Http2Test[jvm] > headersFrameThenContinuation()[jvm] FAILED
    java.lang.ArrayIndexOutOfBoundsException: Index 32 out of bounds for length 32
        at okhttp3.internal.http2.Http2Test.largeHeaders(Http2Test.kt:759)
        at okhttp3.internal.http2.Http2Test.headersFrameThenContinuation(Http2Test.kt:117)
```

**Review by yschimke — APPROVED** [review]
Approved pending the test fixes, assume these are mechanical changes and it's hard to review line for line.

**Comment by swankjesse** [comment]
I was betrayed by ⌘⌥⇧K.

```
   /** Create a sufficiently large header set to overflow INITIAL_MAX_FRAME_SIZE bytes.  */
   private static List<Header> largeHeaders() {
-    String[] nameValues = new String[32];
-    char[] chars = new char[512];
-    for (int i = 0; i < nameValues.length; ) {
-      Arrays.fill(chars, (char) i);
-      nameValues[i++] = nameValues[i++] = String.valueOf(chars);
-    }
-    return headerEntries(nameValues);
-  }
+  private fun largeHeaders(): List<Header> {
+    val nameValues = arrayOfNulls<String>(32)
+    val chars = CharArray(512)
+    var i = 0
+    while (i < nameValues.size) {
+      Arrays.fill(chars, i.toChar())
+      nameValues[i++] = String(chars)
+      nameValues[i++] = nameValues[i++]
+    }
+    return headerEntries(*nameValues)
  }
```

File: okhttp-tls/src/test/java/okhttp3/tls/HandshakeCertificatesTest.kt [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    val serverHandshake = serverHandshakeFuture.get()
+    assertThat(listOf(clientCertificate.certificate, clientIntermediate.certificate))
+      .isEqualTo(serverHandshake.peerCertificates)
+    assertThat(listOf(serverCertificate.certificate, serverIntermediate.certificate))
+      .isEqualTo(serverHandshake.localCertificates)
+    val clientHandshake = clientHandshakeFuture.get()
+    assertThat(listOf(serverCertificate.certificate, serverIntermediate.certificate))
+      .isEqualTo(clientHandshake.peerCertificates)
+    assertThat(listOf(clientCertificate.certificate, clientIntermediate.certificate))
+      .isEqualTo(clientHandshake.localCertificates)
```
Comment (Goooler) [inline okhttp-tls/src/test/java/okhttp3/tls/HandshakeCertificatesTest.kt:99]: Why not use `assertEquals` extension?
```kotlin
fun <@OnlyInputTypes T> assertEquals(expected: T, actual: T, message: String? = null) {
    asserter.assertEquals(message, expected, actual)
}
```

Comment (swankjesse) [reply okhttp-tls/src/test/java/okhttp3/tls/HandshakeCertificatesTest.kt:99]: I prefer fluent asserts with the actual first. 

Might need to migrate to kotlin.test asserts which aren't as nice to finish multiplatform, but that's out of scope here.

**Comment by Goooler** [comment]
Can I convert tests under `okhttp/src/jvmTest/java/okhttp3/internal/http2`?

**Comment by maldito-dot** [comment]
mockwebserver/src/test/java/mockwebserver3/internal/http2/Http2Server.java

File: okhttp-tls/src/test/java/okhttp3/tls/HandshakeCertificatesTest.kt (hunk 1 above)
Comment (Goooler) [reply okhttp-tls/src/test/java/okhttp3/tls/HandshakeCertificatesTest.kt:99]: Alright.

**Pull request merged**